skymp5-server/cpp/lib_espm/espm.cpp

```diff
@@ -1058,3 +1060,62 @@ espm::GMST::Data espm::GMST::GetData(
     compressedFieldsCache);
   return result;
 }
+
+espm::Effects::Effects(const RecordHeader* parent)
+  : parent(parent)
+{
+}
+
+espm::Effects::Data espm::Effects::GetData(
+  CompressedFieldsCache& compressedFieldsCache) const noexcept
+{
+  if (!parent) {
+    return Data();
+  }
+
+  Data result;
+  uint32_t effect = 0;
+
+  espm::RecordHeaderAccess::IterateFields(
+    parent,
+    [&](const char* type, uint32_t size, const char* data) {
+      if (!memcmp(type, "EFID", 4)) {
+        result.effects.emplace_back();
+        result.effects[effect].effectId =
+          *reinterpret_cast<const uint32_t*>(data);
+      } else if (!memcmp(type, "EFIT", 4)) {
+        Effect& eff = result.effects[effect];
```

Comment:
For a vector, this is unsafe. For a map, this can result in filling an item that wasn't prefilled with id. If the order where `EFIT` goes before `EFID` is not possible, I'd suggest adding a check and throwing. If not, maybe it worth adding a post-check that every map item is fully filled